src/resolve/resolved-dns-packet.h

```diff
@@ -35,11 +35,11 @@ struct DnsPacketHeader {
 };
 
 #define DNS_PACKET_HEADER_SIZE sizeof(DnsPacketHeader)
-#define UDP_PACKET_HEADER_SIZE (sizeof(struct iphdr) + sizeof(struct udphdr))
+#define UDP4_PACKET_HEADER_SIZE (sizeof(struct iphdr) + sizeof(struct udphdr))
+#define UDP6_PACKET_HEADER_SIZE (40 + sizeof(struct udphdr))
```

Comment:
sizeof(struct ipv6hdr) ? I know it's fixed at 40, but magic numbers are not the best :-)

Reply:
So, the reason was that `struct ipv6hdr` is defined in Linux headers, that conflict in its definitions with the stuff from `netinet/*.h`. However, I looked close now and found that there's an alternative `struct ip6_hdr` which is defined in `netinet/*.h`, which we can use. Will rework.